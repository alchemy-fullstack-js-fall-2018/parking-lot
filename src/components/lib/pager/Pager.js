import React from 'react';
import PropTypes from 'prop-types';

import styles from './Pager.css';

function Pager({ currentPage, totalPages, updatePage }) {

  const page = (currentPage) ? currentPage : 0;

  let previousButton = null;
  if(page > 0) {
    previousButton =
      <button
        className={styles.pageButton}
        onClick={() => updatePage(page - 1)}
      >
        Previous
      </button>;
  }

  let nextButton = null;
  if(page < totalPages - 1) {
    nextButton =
      <button
        className={styles.pageButton}
        onClick={() => updatePage(page + 1)}
      >
        Next
      </button>;
  }

  return (
    <div>
      {previousButton}
      <span>{page + 1} out of {totalPages} pages</span>
      {nextButton}
    </div>
  );
}

Pager.propTypes = {
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  updatePage: PropTypes.func.isRequired
};

export default Pager;
