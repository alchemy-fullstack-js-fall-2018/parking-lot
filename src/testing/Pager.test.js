import React from 'react';
import { shallow } from 'enzyme';
import Pager from '../components/lib/Pager.jsx';

describe('<Pager />', () => {
  test('renders', () => {
    const wrapper = shallow(<Pager />);
    expect(wrapper).toMatchSnapshot();
  });
});
