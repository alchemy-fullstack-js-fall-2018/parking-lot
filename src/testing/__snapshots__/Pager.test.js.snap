// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pager /> renders 1`] = `
<div>
  <span>
    1
     out of 
     pages
  </span>
</div>
`;
